### PR TITLE
Add a TimeRange class for handling tstzrange types

### PR DESCRIPTION
**Pull request by lassemaatta**
- Wraps a Range<Instant> instance

---
This change is [Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6)

### CODE REVIEW

**Review by lassemaatta — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6)* status: 0 of 9 files reviewed, all discussions resolved (waiting on @lassemaatta)

___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java, line 93 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbKpGHn0ZPmbCuSH_Rw:-MbKpGHn0ZPmbCuSH_Rx:bry4x7e) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java#L93)):*
> ```Java
>         }
>         final Range<Instant> range = u.range();
>         final StringBuilder sb = new StringBuilder(64);
> ```

FYI: In a later commit I will add a similar converter for `LocalDate`s and in that commit this `StringBuilder` stuff is moved to a common utility function as both converters need it.

**Review by jarkkoka — COMMENTED** [review]
> Reviewed 9 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6)* status: all files reviewed, 10 unresolved discussions (waiting on @lassemaatta)

___
*[pom.xml, line 158 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbKofg3Dp0_aODpAJcZ:-MbKofg3Dp0_aODpAJc_:b-f5kecw) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/pom.xml#L158)):*
<details><summary><i alt="summary">Quoted 6 lines of code…</i></summary>

> ```
>         <dependency>
>             <groupId>org.quicktheories</groupId>
>             <artifactId>quicktheories</artifactId>
>             <version>${quicktheories.version}</version>
>             <scope>test</scope>
>         </dependency>
> ```
</details>


No prior experience from this library but I think this property-based testing makes much sense by improving code coverage as well as giving more thought to test code.
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRange.java, line 11 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbKpEoF1Tox2nfMFmsa:-MbKpEoF1Tox2nfMFmsb:b-5z4wx8) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRange.java#L11)):*
> ```
> public abstract Range<Instant> range()
> ```

Ok, there is no equivalent in java.time package for JodaTime-Interval. However, ThreeTen library nowadays has one (in its threeten-extra artifact). This custom implementation is neat so no problem with it. We may later consider using ThreeTen if need be.
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeBinding.java, line 19 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbKrCKy3r01RRFgr30A:-MbKrCKy3r01RRFgr30B:bo91bpc) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeBinding.java#L19)):*
> ```
> Binding<Object, TimeRange>
> ```

I'm not a jOOQ expert so I barely am able to question much in bindings files. However, to my eye key things seem to be right and by comparing to other bindings the differences are what I expected.
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeBinding.java, line 45 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbKuT3OBKJXun1Qi10m:-MbKuT3OBKJXun1Qi10n:bli9kvw) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeBinding.java#L45)):*
> ```
>     // Getting a String value from a JDBC CallableStatement and converting that to a String
>     @Override
> ```

Good comment!
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeUtil.java, line 19 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbKv7i81O6ejjyzcueo:-MbKv7i81O6ejjyzcuep:bsrmit7) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeUtil.java#L19)):*
> ```
>         if (from.isPresent()) {
> ```

Personally, I would prefer functional programming style of using map, flatMap and orElseGet methods (instead of using isPresent and orElseThrow) which would eliminate those redundant orElseThrow calls. However, I'm not suggesting that you need to change this to more idiomatic style (with lambda expressions) because I'm aware many developers feel that traditional and more verbose declarative way of dealing with functional abstraction classes is easier to reason about. In this form, the code is finely readable.
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterGenerativeTest.java, line 16 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLSuAX4Dmw_dpsh1QO:-MbLSuAY9Ylh8unSFwSf:b-bzi2kh) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterGenerativeTest.java#L16)):*
> ```
> TimeRangeConverterGenerativeTest
> ```

Like this approach that augmenting property based testing is done separately in its own class.
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterGenerativeTest.java, line 22 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLU2rYBKu2EmeDQ04y:-MbLU2rYBKu2EmeDQ04z:bsfjq6d) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterGenerativeTest.java#L22)):*
> ```
> Generators
> ```

Nice generators!
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 12 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLDOHF7yDxdk20XbiO:-MbLDOHF7yDxdk20XbiP:b-asxvas) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L12)):*
> ```
> TimeRangeConverterTest
> ```

Nicely compact but at the same time covering most relevant cases!

Since there is no documentation (in form of examples) with regard to valid input formats in the class under test (except the regular expressions itself), the invalid/unaccepted (yet commonly used) timestamp formats remain uncovered. However, a reader may conclude that the formats untested in this class will not pass. E.g. a T character between date and time or missing timezone represent invalid input formats. Testing is done in a kind of practical minimalism fashion. I think it is fine since this project is not meant to be a general purpose library.
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 45 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLDpBt0BpfmbhqxUBG:-MbLDpBt0BpfmbhqxUBH:b-gq3n64) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L45)):*
> ```
>                    is(TimeRange.of(Range.closed(Instant.parse("2004-10-19T08:23:54.00Z"),
>                                                 Instant.parse("2005-10-19T08:23:54.00Z")))));
> ```

The two string literals appear multiple times in this class. I would have declared them as static variables in order to not need to check each appearance for typos. However, I understand this is a matter of taste. So you may or may not make changes based on this remark.
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 75 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLL4ePBNWNEtxSBBLD:-MbLL4eQ3idCJF6LaIX0:bbw7fjr) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L75)):*
> ```
>         assertThat(CONVERTER.from("[\"2021-01-28 13:40:04.392588+02\",)"),
>                    is(TimeRange.of(Range.atLeast(Instant.parse("2021-01-28T11:40:04.392588Z")))));
> ```

Good that quotes are tested at least somewhere. Start/end quotes are not needed to match either. In other words a timestamp may have a start quote without an end quote. But quotes are discarded anyway in the transformation.

**Review by tk-vincit — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6)* status: all files reviewed, 14 unresolved discussions (waiting on @jarkkoka and @lassemaatta)

___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java, line 62 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLOZiL30pAPzqYWFNI:-MbLOZiL30pAPzqYWFNJ:bu2d932) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java#L62)):*
> ```Java
>         final Matcher m = PATTERN.matcher(t.toString()
>                                            .replaceAll("\"", ""));
>         if (m.find()) {
> ```

I don't know anything about Java's Matcher type, but should we check something like `if (m.size == 4) { ... } else { throw Error(...) }`?
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java, line 93 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLPQ7l-KxjlbczOa-h:-MbLPQ7l-KxjlbczOa-i:b-gzs7h3) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java#L93)):*
> ```Java
>         }
>         final Range<Instant> range = u.range();
>         final StringBuilder sb = new StringBuilder(64);
> ```

Is there a reason why variable names are like "u" or "sb"? This would be easier to read if those would have meaningful names instead. Or are these names self-evident to seasoned Java developers?
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java, line 110 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLQaL-0t-_nIzPcJ9r:-MbLQaL-0t-_nIzPcJ9s:b465pn6) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java#L110)):*
> ```Java
>         if (range.hasUpperBound()) {
>             sb.append(toString(range.upperEndpoint()));
>             switch (range.upperBoundType()) {
> ```

Below on `TimeRangeUtil.java` similar conditions had been implemented with simple ternary operator. Switch-case is of course easier to extend if we start living in a world where ranges can have new kinds of bounds and thus more elegant - but should we still strive for consistency and use same operators (switch or ternary, both work for me) in both places?
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 36 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLT6Mb9Sl9wRZkIbas:-MbLT6Mb9Sl9wRZkIbat:bzi0xrw) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L36)):*
> ```Java
>         assertThat(range,
>                    is(TimeRange.unbounded()));
>         final TimeRange range2 = CONVERTER.from("[,]");
> ```

This is not meaningful name. One way to avoid naming would be inlining this into assertion. Anyway, this is OK for me if you prefer this approach.
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 45 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLDpBt0BpfmbhqxUBG:-MbLUjnb04ungWE_rfL9:baei46w) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L45)):*
<details><summary><i>Previously, jarkkoka (Jarkko Kaura) wrote…</i></summary><blockquote>

> ```
>                    is(TimeRange.of(Range.closed(Instant.parse("2004-10-19T08:23:54.00Z"),
>                                                 Instant.parse("2005-10-19T08:23:54.00Z")))));
> ```

The two string literals appear multiple times in this class. I would have declared them as static variables in order to not need to check each appearance for typos. However, I understand this is a matter of taste. So you may or may not make changes based on this remark.
</blockquote></details>

I was drafting a bit similar comment... This kind of date strings are pretty hard to parse for humans, so these tests are quite difficult to read. As a pseudo code, you could do something like this:
```

final Date START_DATE = Date.fromISOString(...)
final Date END_DATE =  Date.fromISOString(...)

function buildOpenDateRange (start: Date, end: Date) {
    return `(${start.date.toString()${start.time.toString}, ...}`
}

final string DATE_TO_PARSE = buildOpenDateRange(START_DATE, END_DATE)
assertThat(CONVERTER.from(DATE_TO_PARSE), is(timeRange.of(Range.closed(START_DATE, END_DATE))))
```

**Review by lassemaatta — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6)* status: all files reviewed, 14 unresolved discussions (waiting on @jarkkoka, @lassemaatta, and @tk-vincit)

___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeBinding.java, line 19 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbKrCKy3r01RRFgr30A:-MbLYncxC0EatLSfrgmp:b-yn0v3z) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeBinding.java#L19)):*
<details><summary><i>Previously, jarkkoka (Jarkko Kaura) wrote…</i></summary><blockquote>

> ```
> Binding<Object, TimeRange>
> ```

I'm not a jOOQ expert so I barely am able to question much in bindings files. However, to my eye key things seem to be right and by comparing to other bindings the differences are what I expected.
</blockquote></details>

Neither am I. There's not much documentation about `Binding`s and `Converter`s and I'm mostly relying on sample code found online.
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java, line 62 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLOZiL30pAPzqYWFNI:-MbLZFg1BSWyxewG_SEJ:b-vs8h1d) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java#L62)):*
<details><summary><i>Previously, tk-vincit wrote…</i></summary><blockquote>

I don't know anything about Java's Matcher type, but should we check something like `if (m.size == 4) { ... } else { throw Error(...) }`?
</blockquote></details>

It's probably redundant, as `m.group(N)` throws if there's no such group (ie. the regex and the code do not match).
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java, line 93 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLPQ7l-KxjlbczOa-h:-MbLZhuI9i_IUHkzm0OS:b-pll3pr) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java#L93)):*
<details><summary><i>Previously, tk-vincit wrote…</i></summary><blockquote>

Is there a reason why variable names are like "u" or "sb"? This would be easier to read if those would have meaningful names instead. Or are these names self-evident to seasoned Java developers?
</blockquote></details>

I think the `t` and `u` parameters originate from the fact that the `Converter` class generic signature is  `Converter<T,​U>`, where `t/T` is the database type and `u/U` is the user type.

I've often seen `StringBuilder` instances shortened as  `sb`, but I'm not sure whether it's a common convention.
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java, line 110 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLQaL-0t-_nIzPcJ9r:-MbL_IBN6B3Y_XHD9_v3:b54g86d) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java#L110)):*
<details><summary><i>Previously, tk-vincit wrote…</i></summary><blockquote>

Below on `TimeRangeUtil.java` similar conditions had been implemented with simple ternary operator. Switch-case is of course easier to extend if we start living in a world where ranges can have new kinds of bounds and thus more elegant - but should we still strive for consistency and use same operators (switch or ternary, both work for me) in both places?
</blockquote></details>

I'm pretty sure (I'll verify) that you can't use ternary expressions as control flow constructs. That is, it's valid to say `return isFoo ? 1 : 2;` or `final int foo = isFoo ? 1 : 2;`, but you can't say `isFoo ? myFunc() : otherFunc();`
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeUtil.java, line 19 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbKv7i81O6ejjyzcueo:-MbLXghz4Ofw5LA2qDoT:bmvoikw) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeUtil.java#L19)):*
<details><summary><i>Previously, jarkkoka (Jarkko Kaura) wrote…</i></summary><blockquote>

> ```
>         if (from.isPresent()) {
> ```

Personally, I would prefer functional programming style of using map, flatMap and orElseGet methods (instead of using isPresent and orElseThrow) which would eliminate those redundant orElseThrow calls. However, I'm not suggesting that you need to change this to more idiomatic style (with lambda expressions) because I'm aware many developers feel that traditional and more verbose declarative way of dealing with functional abstraction classes is easier to reason about. In this form, the code is finely readable.
</blockquote></details>

I agree and generally I prefer the functional style. However, I remember that I've previously tried to implement a "I have multiple `Optional<Foo>` things and I need to do something depending on which are present" code and noticed that it's really hard to figure the proper nesting of multiple `flatMap` + `orElse*` calls and, naturally, I also find it rather hard to read & reason about.

**Review by lassemaatta — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6)* status: all files reviewed, 13 unresolved discussions (waiting on @jarkkoka, @lassemaatta, and @tk-vincit)

___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 12 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLDOHF7yDxdk20XbiO:-MbZxLAxCRIFjKHtEeWx:b-sfap59) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L12)):*
<details><summary><i>Previously, jarkkoka (Jarkko Kaura) wrote…</i></summary><blockquote>

> ```
> TimeRangeConverterTest
> ```

Nicely compact but at the same time covering most relevant cases!

Since there is no documentation (in form of examples) with regard to valid input formats in the class under test (except the regular expressions itself), the invalid/unaccepted (yet commonly used) timestamp formats remain uncovered. However, a reader may conclude that the formats untested in this class will not pass. E.g. a T character between date and time or missing timezone represent invalid input formats. Testing is done in a kind of practical minimalism fashion. I think it is fine since this project is not meant to be a general purpose library.
</blockquote></details>

This converter specifically targets the Postgresql `tstzrange` class, which contains datetime strings in the ISO 8601 format. I added some comments regarding this but also added a few failing tests cases to indicate that other date/time formats are not supported.
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 36 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLT6Mb9Sl9wRZkIbas:-MbZzGEk0ahTlWzsS267:bb6un1j) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L36)):*
<details><summary><i>Previously, tk-vincit wrote…</i></summary><blockquote>

This is not meaningful name. One way to avoid naming would be inlining this into assertion. Anyway, this is OK for me if you prefer this approach.
</blockquote></details>

Done
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 45 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLDpBt0BpfmbhqxUBG:-MbZvHfJ8M9KR61VC-Cm:bb6un1j) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L45)):*
<details><summary><i>Previously, tk-vincit wrote…</i></summary><blockquote>

I was drafting a bit similar comment... This kind of date strings are pretty hard to parse for humans, so these tests are quite difficult to read. As a pseudo code, you could do something like this:
```

final Date START_DATE = Date.fromISOString(...)
final Date END_DATE =  Date.fromISOString(...)

function buildOpenDateRange (start: Date, end: Date) {
    return `(${start.date.toString()${start.time.toString}, ...}`
}

final string DATE_TO_PARSE = buildOpenDateRange(START_DATE, END_DATE)
assertThat(CONVERTER.from(DATE_TO_PARSE), is(timeRange.of(Range.closed(START_DATE, END_DATE))))
```
</blockquote></details>

Done
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 75 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLL4ePBNWNEtxSBBLD:-MbZvJn45EDXG6NiYrlM:bb6un1j) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L75)):*
<details><summary><i>Previously, jarkkoka (Jarkko Kaura) wrote…</i></summary><blockquote>

> ```
>         assertThat(CONVERTER.from("[\"2021-01-28 13:40:04.392588+02\",)"),
>                    is(TimeRange.of(Range.atLeast(Instant.parse("2021-01-28T11:40:04.392588Z")))));
> ```

Good that quotes are tested at least somewhere. Start/end quotes are not needed to match either. In other words a timestamp may have a start quote without an end quote. But quotes are discarded anyway in the transformation.
</blockquote></details>

Done

**Review by jarkkoka — APPROVED** [review]
> Reviewed 3 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6)* status: 8 of 9 files reviewed, 14 unresolved discussions (waiting on @jarkkoka, @lassemaatta, and @tk-vincit)

___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java, line 23 at r2](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-Mb_CI_qCPpvjh1i1jAq:-Mb_CI_r9OD9VBYflORj:b-ssj31g) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/9cc134d3e5b5356c8005835426df5a78852e8187/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java#L23)):*
<details><summary><i alt="summary">Quoted 7 lines of code…</i></summary>

> ```
> /**
>  * A converter class for converting between:
>  * <ul>
>  *     <li>Postgresql {@code tstzrange} values in the ISO 8601 format (e.g. {@code "[2004-10-19 10:23:54+02,2005-10-19 10:23:54+02)")} and
>  *     <li>{@link fi.hsl.jore.importer.config.jooq.converter.time_range.TimeRange} instances
>  * </ul>
>  */
> ```
</details>


Nice.
___
*[src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java, line 42 at r2](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-Mb_9JFO5kSQkYWyCWFm:-Mb_9JFO5kSQkYWyCWFn:bjxdus8) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/9cc134d3e5b5356c8005835426df5a78852e8187/src/main/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverter.java#L42)):*
<details><summary><i alt="summary">Quoted 5 lines of code…</i></summary>

> ```
>     /**
>      * We cant use {@link java.time.format.DateTimeFormatter#ISO_LOCAL_DATE_TIME ISO_LOCAL_DATE_TIME} directly,
>      * as "ISO 8601 specifies the use of uppercase letter T to separate the date and time. PostgreSQL accepts
>      * that format on input, but on output it uses a space rather than T."
>      */
> ```
</details>


Well clarified.
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 75 at r1](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-MbLL4ePBNWNEtxSBBLD:-Mb_9sHo5l_YxbVKfTZD:by65r6r) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/5006167150387e45a10873b3a385e2e69055b914/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L75)):*
<details><summary><i>Previously, lassemaatta (Lasse Määttä) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Clear improvement!
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 52 at r2](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-Mb_BhyH4ZWrKH92TiYW:-Mb_BhyI2G1dH1sSCkXw:b-f0ipmr) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/9cc134d3e5b5356c8005835426df5a78852e8187/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L52)):*
> ```
> givenBounded_thenReturnBounded
> ```

Only (pretty minor) thing missing from this method is to test those closedOpen/openClosed cases. Adding that would be such a trivial thing to do that it wouldn't warrant a new review cycle IMHO.
___
*[src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java, line 98 at r2](https://reviewable.io/reviews/hsldevcom/jore4-jore3-importer/6#-Mb_BGcOBMUgBpxTwq9C:-Mb_BGcOBMUgBpxTwq9D:brbewmr) ([raw file](https://github.com/hsldevcom/jore4-jore3-importer/blob/9cc134d3e5b5356c8005835426df5a78852e8187/src/test/java/fi/hsl/jore/importer/config/jooq/converter/time_range/TimeRangeConverterTest.java#L98)):*
> ```
> givenStringWithQuotes_thenReturnParsed()
> ```

Nice addition to detail.
